lib/src/results.dart

```diff
@@ -135,6 +137,99 @@ extension RealmResultsOfObject<T extends RealmObjectBase> on RealmResults<T> {
   }
 }
 
+class _SubscribedRealmResult<T extends RealmObject> extends RealmResults<T> {
+  final String? subscriptionName;
+
+  _SubscribedRealmResult._(RealmResults<RealmObject> results, {this.subscriptionName})
+      : super._(
+          results.handle,
+          results.realm,
+          results.metadata,
+        );
+}
+
+extension RealmResultsOfRealmObject<T extends RealmObject> on RealmResults<T> {
```

Comment:
Because we allow making queries with embedded objects to a local realm. But it is not allowed to make subscriptions on embedded objects for sync realm.